docs/guidelines/platforms/android/guideline_robust_android.md

```diff
@@ -1,94 +1,142 @@
-
-In these situations, it is recommended to use `android:hint` attribute.
-
-**Code example:**
+Example: setting a click action with description to card:
 
 ```
-<!-- The hint text for en-US locale would be
-     "Password". -->
-<EditText
-   android:id="@+id/passwordInputField"
-   android:hint="@string/password_hint_text" ... />
+<!-- Label text for en-US locale would be "Open settings" -->
+ Card(
+    modifier = Modifier
+        .semantics {
+            onClick(
+             label = openSettingsLabel,
+                action = {
+                    openSettings() // action to be performed
+                    true           // return true to indicate that the action was handled
+                }
+            )
+        }
+ ) {
+  ...
+ }
 ```
 
+For users of switch control, actions like drag-and-drop can be challenging. To make this functionality more accessible, custom action can be
+defined.
 
-- Elements in the collection
+Example: setting a custom action for drag-and-drop to add favourites:
 
-When working with collections, it is very important that provided labels are **unique for each item**. That way, the system accessibility services can refer to exactly one specific item when announcing the label.
+```
+<!-- Label text for en-US locale would be "Add to favourites" -->
+ Card(
+    modifier = Modifier
+        .semantics {
+            customActions = listOf(
+                // your custom action
+                CustomAccessibilityAction(label = addToFavouritesLabel) {
+                    // Add to favourite logic
+                    true
+                }
+            )
+        }
+ ) {
+  ...
+ }
+```
 
-This will help users to get a general idea when they've cycled through the complete UI of your app.
+To help users understand the purpose of an element, roles should be added to the elements.
+In View system, roles can be defined through `AccessibilityDelegateCompat` with `setRoleDescription`, however, using `setClassName` should
+be preferred as it is multilingual by default.
 
-In particular, you should provide additional text or contextual information in elements within reused layouts, such as RecyclerView objects, so each child element is uniquely identified.
+Example: setting button role to a card view
 
-The described requests could be achieved by setting content descriptions as a part of adapter implementation.
+```
+ ViewCompat.setAccessibilityDelegate(
+        loginCard,
+        object : AccessibilityDelegateCompat() {
+            override fun onInitializeAccessibilityNodeInfo(host: View, info: AccessibilityNodeInfoCompat) {
+                super.onInitializeAccessibilityNodeInfo(host, info)
+                info.roleDescription = "Button"
+                info.className = Button::class.java.name
```

Comment:
Just curious, how is this read then via TalkBack?

Reply:
If we had a card which said login, the TalkBack would read "Button, Login".